tests/ttnn/unit_tests/operations/test_conv1d.py

```diff
@@ -120,6 +120,8 @@ def run_conv(
         conv_op_cache=reader_patterns_cache,
         debug=debug,
         groups=groups,
+        return_output_length=True,
+        return_prepared_device_weights=True,
```

Comment:
What's the benefit of giving the user these options vs. just doing what we currently do and return everything?

IMO these flags make the API more error-prone, and I can't see why there would be a significant performance implication of enabling vs. disabling these features.

Reply:
This is something to take up with @mywoodstock and @pavlejosipovic. Us three thought this would be best since normal op behaviour is to return one output tensor. If the user wishes to have the additional output they need to explicitly state that.

I’m not on this project I’m just doing this because MLIR needs to model weight preparation that doesn’t include running the conv2d op first. If API design specific things are at issue Im not really at liberty to make a decision.

Reply:
If it was up to me conv would return just output tensor, and if user wants to have prepared bias/weights they would have to call separate function first, but I've accepted this as compromise.

Reply:
@esmalTT yeah, this update is for the requirements for the forge/MLIR effort, not for performance.

Reply:
> If it was up to me conv would return just output tensor, and if user wants to have prepared bias/weights they would have to call separate function first, but I've accepted this as compromise.

Yeah this would be the best, since it is the simplest.